### DISCUSSION TITLE
How to get nested relation in t.crud

### DISCUSSION THREAD

**Discussion by aakashkag** [prisma/prisma#2870] [Q&A] [answered]
Hi,
I want to get nested relation.

Let's say i have two table user and post.
```prisma
model User {
  email String  @unique
  id    Int     @default(autoincrement()) @id
  name  String?
  posts Post[]
}

model Post {
  authorId  Int?
  content   String?
  id        Int     @default(autoincrement()) @id
  published Boolean @default(false)
  title     String
  author    User?   @relation(fields: [authorId], references: [id])
}

```
I am exposing ready api like this. 

```ts
 t.crud.users({
            filtering: true,
            ordering: true,
            pagination: true,
})

```
Now I would like to get user along with there posts.
I am trying to write.

```graphql
query{
  users(where:{id:{in:[1,2]}}){

    }
)

```

but how to get posts in users query?

**Comment by ryands17** [comment, marked as answer]
@aakashkag You need to create the User and Post model in Nexus as follows:

```ts
const Post = objectType({
  name: 'Post',
  definition(t) {
    t.model.id()
    t.model.title()
    t.model.content()
  },
})

const User = objectType({
  name: 'User',
  definition(t) {
    t.model.id()
    t.model.name()
    t.model.posts({
      filtering: true,
      pagination: true,
      ordering: true,
    })
  },
})
```

And the query in the following manner:

```graphql
query {
  users(where: { id: { in: [1, 2] } }) {
    id
    name
    posts(orderBy: { title: asc }) {
      title
      content
    }
  }
}
```

**Reply by ryands17** [reply]
Hey @aakashkag :wave: 
Did you get the above to work?

**Comment by aakashkag** [comment]
Hello @ryands17 
It's working fine except self relationship. 
I have self relationship in user table.
 Would like to expose something like this

```ts
const User = objectType({
  name: 'User',
  definition(t) {
    t.model.id()
    t.model.name()
    t.model.user{
      filtering: true,
      pagination: true,
      ordering: true,
    })
  },
})
```

**Reply by ryands17** [reply]
@aakashkag It's working for me in self relations as well. I have created a self relation in the following manner:

```prisma
model User {
  id        Int      @default(autoincrement()) @id
  email     String   @unique
  name      String?
  following User[]   @relation("follows")
  followers User[]   @relation("follows")
}
```

And then I have added this to Nexus as follows:

```ts
const User = objectType({
  name: 'User',
  definition(t) {
    t.model.id()
    t.model.email()
    t.model.name()
    t.model.followers({
      pagination: true,
    })
    t.model.following({
      pagination: true,
    })
  },
})

const Query = queryType({
  definition(t) {
    t.crud.users()
  },
})
```

And finally this is the GraphQL Query:

```graphql
query {
  users {
    id
    name
   	followers {
      name
    }
    following {
      name
    }
  }
}
```

Let me know if this works or you could share your User schema if possible.

**Reply by aakashkag** [reply]
Thanks for response.
Can't we apply filter and ordering in self relation.?

**Reply by ryands17** [reply]
Yes ordering and filtering both will work properly :)

**Comment by ludralph** [comment]
Hi @aakashkag,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
